move/example/sources/its/its.move

```diff
@@ -0,0 +1,177 @@
+module example::its {
+    use std::ascii;
+    use std::ascii::{String};
+    use std::type_name;
+
+    use sui::event;
+    use sui::address;
+    use sui::hex;
+    use sui::coin::{Coin};
+    use sui::sui::SUI;
+    use sui::clock::Clock;
+
+    use axelar_gateway::channel::{Self, Channel, ApprovedMessage};
+    use axelar_gateway::discovery::{Self, RelayerDiscovery, Transaction};
+
+    use gas_service::gas_service::GasService;
+
+    use its::service;
+    use its::its::ITS;
+    use its::token_id::TokenId;
+
+    public struct Singleton has key {
+        id: UID,
+        channel: Channel,
+    }
+
+    public struct Executed has copy, drop {
+        source_chain: String,
+        source_address: vector<u8>,
+        data: vector<u8>,
+        amount: u64,
+    }
+
+    fun init(ctx: &mut TxContext) {
+        let singletonId = object::new(ctx);
+        let channel = channel::new(ctx);
+        transfer::share_object(Singleton {
+            id: singletonId,
+            channel,
+        });
+    }
+
+    public fun register_transaction(discovery: &mut RelayerDiscovery, singleton: &Singleton, its: &ITS) {
+        let mut arguments = vector::empty<vector<u8>>();
+
+        // Singleton object
+        let mut arg = vector::singleton<u8>(0);
```

Comment:
perhaps we create a module somewhere to store all possible argument types?

for example,

```move
module gmp::arg_types {
public const TYPE_OBJECT u8 = 0;
/// Prefix for pure arguments, followed by the BCS encoded form of the pure
public const TYPE_PURE u8 = 1;
/// ... and so on
```

then import and use it here like below:
```move
use gmp::arg_types;

create_argument(arg_types::TYPE_OBJECT, singleton)
```